### ISSUE TITLE
Extend support of dynamic icon

### ISSUE THREAD

**Issue by Fejitatete** [openhab/openhab-android#3461]
Is there a possibility to extend the support of dynamic icon, like basicui.
For example : the app doesn't care about custom dynamic icon in text label (just ON / OFF, see pictures attached).

1st picture : basicui
![basicui](https://github.com/openhab/openhab-android/assets/6900432/797e0967-6264-4dab-92de-4c7f8a0088bc)
2nd picture : android app
![openhab_android](https://github.com/openhab/openhab-android/assets/6900432/45b26d68-4d48-43d8-b1cf-684889f2b0c3)

Thank you for your work & reading.
Cheers.

**Comment by maniac103** [comment]
What do you mean by 'custom dynamic icon'? Can you please show item and sitemap definition?
In general the app just includes the item state when requesting the icon, and it's up to the server to deliver the correct icon. If there's a mismatch between app and BasicUI, it must thus be inside the state sending code, and for that the item and sitemap widget definition is important.

**Comment by mueller-ma** [comment]
Do you have the battery saver or data saver enabled? What's the value of `loadIconsWithState` in the log of the app in line ~10?

**Comment by maniac103** [comment]
OK, so I guess the dynamic icons for ON and OFF work fine and the issue is just with the '-' state? 
(And if the answer is yes: what kind of state is this even, considering it's neither NULL nor UNDEF and Switch items can not get arbitrary string states?)

**Comment by Fejitatete** [comment]
> OK, so I guess the dynamic icons for ON and OFF work fine and the issue is just with the '-' state?

Correct

>What binding does this state come from?

It's a rule that postUpdate 'NULL' to the item.
AFAIK, all item with 'NULL' state are displayed '-' in basicui.

**Comment by Fejitatete** [comment]
Sorry, my mistake : the item is displayed as '-' so with the transform it become 'Hors Ligne' but the icon shown in basicui is 'light-null.png'

rule :
```
/* global Java, event, items, actions */ 
(function () {
  const THING_ID = 'miio:generic:0EEA652E'
  const OUTPUT_ITEM_NAME = 'MiSmartLEDSalon_isOnline'

  const thingTurnedOffline = typeof event !== 'undefined' && event.statusInfo !== undefined && event.statusInfo.status.toString() === 'OFFLINE'
  const thingTurnedOnline = typeof event !== 'undefined' && event.statusInfo !== undefined && event.statusInfo.status.toString() === 'ONLINE'

  // handle each case separately
 if (thingTurnedOffline) {
    console.debug(THING_ID + ' turned OFFLINE -> Considering non-alive (' + OUTPUT_ITEM_NAME + ')')
    items.getItem(OUTPUT_ITEM_NAME).postUpdate('NULL')
  } else if (thingTurnedOnline) {
    console.debug(THING_ID + ' turned ONLINE -> Considering alive (' + OUTPUT_ITEM_NAME + ')')
    items.getItem(OUTPUT_ITEM_NAME).postUpdate('ON')
  } else {
    //console.error('Unexpected condition')
    console.debug(THING_ID + ' turned OFFLINE -> Considering non-alive (' + OUTPUT_ITEM_NAME + ')')
    items.getItem(OUTPUT_ITEM_NAME).postUpdate('NULL')
  }
})()
```

**Comment by maniac103** [comment]
Okay, so the issue is just the NULL state. I'll have a look at it.

**Comment by Fejitatete** [comment]
Thank you very much.

**Comment by mueller-ma** [comment]
Closed by  #3456
